internal/srv/handlers.go

```diff
@@ -80,32 +86,40 @@ func (s *Server) processChange(messages <-chan *message.Message) {
 			} else {
 				lb, err = s.newLoadBalancer(m.SubjectID, m.AdditionalSubjectIDs)
 				if err != nil {
-					s.Logger.Errorw("unable to initialize loadbalancer", "error", err, "messageID", msg.UUID)
-					msg.Nack()
+					s.Logger.Errorw("unable to initialize loadbalancer", "error", err, "messageID", msg.UUID, "subjectID", m.SubjectID.String())
+					if errors.Is(err, lbapi.ErrLBNotfound) {
+						// ack and ignore
+						msg.Ack()
+					} else {
+						// nack and retry
+						msg.Nack()
+					}
+
+					continue
 				}
 			}
 
-			if lb.lbType != typeNoLB {
+			if lb != nil && lb.lbType != typeNoLB {
```

Comment:
Same as above. Should we look to check this and provide a better error message if LB data hasn't been populated.